nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -247,33 +233,33 @@ void Optimizer::generateNoisedTrajectories()
 void Optimizer::applyControlSequenceConstraints()
 {
   auto & s = settings_;
-
-  if (isHolonomic()) {
-    control_sequence_.vy = xt::clip(control_sequence_.vy, -s.constraints.vy, s.constraints.vy);
-  }
-
-  control_sequence_.vx = xt::clip(control_sequence_.vx, s.constraints.vx_min, s.constraints.vx_max);
-  control_sequence_.wz = xt::clip(control_sequence_.wz, -s.constraints.wz, s.constraints.wz);
-
-  float max_delta_vx = s.model_dt * s.constraints.ax_max;
-  float min_delta_vx = s.model_dt * s.constraints.ax_min;
-  float max_delta_vy = s.model_dt * s.constraints.ay_max;
-  float max_delta_wz = s.model_dt * s.constraints.az_max;
-  float vx_last = control_sequence_.vx(0);
-  float vy_last = control_sequence_.vy(0);
-  float wz_last = control_sequence_.wz(0);
-  for (unsigned int i = 1; i != control_sequence_.vx.shape(0); i++) {
+  float & vx_min = s.constraints.vx_min;
+  float & vx_max = s.constraints.vx_max;
+  float & vy = s.constraints.vy;
+  float & wz = s.constraints.wz;
+
+  float && max_delta_vx = s.model_dt * s.constraints.ax_max;
+  float && min_delta_vx = s.model_dt * s.constraints.ax_min;
+  float && max_delta_vy = s.model_dt * s.constraints.ay_max;
+  float && max_delta_wz = s.model_dt * s.constraints.az_max;
+  float vx_last = std::min(vx_max, std::max(control_sequence_.vx(0), vx_min));
+  float vy_last = std::min(vy, std::max(control_sequence_.vy(0), -vy));
+  float wz_last = std::min(wz, std::max(control_sequence_.wz(0), -wz));
+  for (unsigned int i = 1; i != control_sequence_.vx.size(); i++) {
     float & vx_curr = control_sequence_.vx(i);
-    vx_curr = std::clamp(vx_curr, vx_last + min_delta_vx, vx_last + max_delta_vx);
+    vx_curr = std::min(vx_max, std::max(vx_curr, vx_min));
+    vx_curr = std::min(vx_last + max_delta_vx, std::max(vx_curr, vx_last + min_delta_vx));
     vx_last = vx_curr;
 
     float & wz_curr = control_sequence_.wz(i);
-    wz_curr = std::clamp(wz_curr, wz_last - max_delta_wz, wz_last + max_delta_wz);
```

Comment:
Worth having a util for clamp that implements this instead of putting inline for each location? It is prone to copy+paste errors

Reply:
I think this is still nice-to-have, but I won't block for it

Reply:
@Ayush1285 thoughts on adding this? it would make things quite a bit more readable and less error prone for future modifiations

Reply:
We can add a util function. What will it implement? Clamping for all x, y, and z values, or just wrapping std::min(..., std::max())?

Reply:
Just a wrapper around `min(max())`